tests/Unit/src/handlers/LiDARHandler.cc

```diff
@@ -0,0 +1,294 @@
+TEST_F(LiDARHandlerTests, QueryWithTraversalScoreRange)
+{
+    LiDARHandler handler;
+    ASSERT_TRUE(handler.OpenDB(m_szDbPath));
+    LiDARHandler::PointFilter filter{.dEasting        = 614058.84,
+                                     .dNorthing       = 4189968.85,
+                                     .dRadius         = 3.0,
+                                     .dTraversalScore = std::optional<LiDARHandler::PointFilter::Range<double>>({0.95, 1.0})};
+    std::vector<LiDARHandler::PointRow> vResults = handler.GetLiDARData(filter);
+    // Should not crash, may be empty
+    SUCCEED();
+}
+
+/******************************************************************************
+ * @brief Query with all range filters.
+ *
+ *
+ * @author clayjay3 (claytonraycowen@gmail.com)
+ * @date 2025-07-14
+ ******************************************************************************/
+TEST_F(LiDARHandlerTests, QueryWithAllRangeFilters)
+{
+    LiDARHandler handler;
+    ASSERT_TRUE(handler.OpenDB(m_szDbPath));
+    LiDARHandler::PointFilter filter{
+        .dEasting        = 614058.84,
+        .dNorthing       = 4189968.85,
+        .dRadius         = 3.0,
+        .dNormalX        = std::optional<LiDARHandler::PointFilter::Range<double>>({-1.0, 1.0}),
+        .dNormalY        = std::optional<LiDARHandler::PointFilter::Range<double>>({-1.0, 1.0}),
+        .dNormalZ        = std::optional<LiDARHandler::PointFilter::Range<double>>({-1.0, 1.0}),
+        .dSlope          = std::optional<LiDARHandler::PointFilter::Range<double>>({0.0, 90.0}),
+        .dRoughness      = std::optional<LiDARHandler::PointFilter::Range<double>>({0.0, 10.0}),
+        .dCurvature      = std::optional<LiDARHandler::PointFilter::Range<double>>({-10.0, 10.0}),
+        .dTraversalScore = std::optional<LiDARHandler::PointFilter::Range<double>>({0.0, 1.0}),
+    };
+    std::vector<LiDARHandler::PointRow> vResults = handler.GetLiDARData(filter);
+    SUCCEED();
+}
+
+/******************************************************************************
+ * @brief Check thread safety of open/close/query operations.
+ *
+ *
+ * @author clayjay3 (claytonraycowen@gmail.com)
+ * @date 2025-07-14
+ ******************************************************************************/
+TEST_F(LiDARHandlerTests, ThreadSafetyOpenCloseQuery)
+{
+    LiDARHandler handler;
+    auto openClose = [&handler, this]()
+    {
+        for (int i = 0; i < 5; ++i)
+        {
+            handler.OpenDB(m_szDbPath);
+            handler.CloseDB();
+        }
+    };
+    auto query = [&handler, this]()
+    {
+        for (int i = 0; i < 5; ++i)
+        {
+            handler.OpenDB(m_szDbPath);
+            LiDARHandler::PointFilter filter{.dEasting = 614058.84, .dNorthing = 4189968.85, .dRadius = 3.0};
+            handler.GetLiDARData(filter);
+            handler.CloseDB();
+        }
+    };
+    std::thread t1(openClose);
+    std::thread t2(query);
+    t1.join();
+    t2.join();
+    SUCCEED();
```

Comment:
Remove auto type since it's not an authorized type in our standards. And update to using the naming convention for variables.

```suggestion
    LiDARHandler stHandler;
    std::function<void()> fnOpenClose = [&stHandler, this]()
    {
        for (int i = 0; i < 5; ++i)
        {
            stHandler.OpenDB(m_szDbPath);
            stHandler.CloseDB();
        }
    };
    std::function<void()> fnQuery = [&stHandler, this]()
    {
        for (int i = 0; i < 5; ++i)
        {
            stHandler.OpenDB(m_szDbPath);
            LiDARHandler::PointFilter stFilter{.dEasting = 614058.84, .dNorthing = 4189968.85, .dRadius = 3.0};
            stHandler.GetLiDARData(stFilter);
            stHandler.CloseDB();
        }
    };
    std::thread th1(fnOpenClose );
    std::thread th2(fnQuery);
    th1.join();
    th2.join();
    SUCCEED();
```